src/include/network/lan/address.rb

```diff
@@ -1631,5 +1611,36 @@ def repeated_physical_port_ids(slaves)
 
       physical_port_ids
     end
+
+    # Performs hostname update
+    #
+    # @param [String] ip address
+    # @param [String] new hostname
+    def update_hostname(ipaddr, hostname)
+      ip_changed = LanItems.ipaddr != ipaddr
+      hostname_initial = hostname_initial(LanItems.ipaddr)
+      hostname_changed = hostname_initial != hostname
+
+      return if !(ip_changed || hostname_changed || hostname.empty?)
+
+      log.info("Dropping record for #{LanItems.ipaddr} from /etc/hosts")
+
+      Host.remove_ip(LanItems.ipaddr)
+      Host.Update(hostname_initial, hostname, ipaddr) if !hostname.empty?
+
+      nil
+    end
+
+    # Returns canonical hostname for the given ip
+    def hostname_initial(ipaddr)
```

Comment:
1. naming is a bit inconsistent. The first one is `update_hostname` and second is `hostname_initial`.
2. 11 new lines added to already long file
3. file is include, so even worse.
4. both method have hostname in its name. Maybe oppurtinity to start with proper hostname object?

Reply:
ad (1) hm, hostname_initial copied original name of instance variable. I'll change it.
ad (2) just moving lines out of pretty long method, resulting file is in fact approx 3 lines shorter (not counting added docu and some white spaces)
ad (4) may be, but it is maintenance update and I don't want to create another bug by overlooking some nasty side effect.